code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/deployment/ServiceDeploymentPlanner.java

```diff
@@ -205,8 +205,14 @@ public void cleanupUnhealthyUnits() {
             next.remove(false, ServiceDiscoveryConstants.AUDIT_LOG_REMOVE_UNHEATLHY);
             it.remove();
         }
-        for (DeploymentUnit toWatch : watchList) {
-            toWatch.waitForRemoval();
+    }
+
+    public void waitForRemoval() {
```

Comment:
if we are consolidating the wait here, we should probably remove the call to waitForRemoval from cleanupBadUnits/cleanupUnhealthyUnits calls